src/datareader.cpp

```diff
@@ -202,6 +208,31 @@ void DataReader::readFile(QString file)
             m_tracknum = QString::number(tagFile->tag()->track());
 
             if (m_title=="") m_title = QFileInfo(file).baseName();
+
+            // if we have artist and album, we check for a cover image.
+            if (m_artist != "" && m_album != "") {
+                QFileInfo info(file);
+                QDirIterator iterator(info.dir().path(), QDirIterator::Subdirectories);
+                while (iterator.hasNext()) {
+                    iterator.next();
+                    // we are explicit about two common factors,
+                    // suffix jpg or png, basename cover or folder
+                    if (iterator.fileInfo().isFile()) {
+                        if (  ( iterator.fileInfo().suffix() == "jpg"  ||
+                                iterator.fileInfo().suffix() == "jpeg" ||
+                                iterator.fileInfo().suffix() == "png"     )   &&
+                              ( iterator.fileInfo().baseName() == "cover" ||
+                                iterator.fileInfo().baseName() == "folder" )
+                              )
+                        {
+                            QString th2 = QStandardPaths::writableLocation(QStandardPaths::CacheLocation) + "/media-art/album-"+ doubleHash(m_artist, m_album) + ".jpeg";
```

Comment:
If the source file is a PNG file, the extension should be png. You may reuse the iterator suffix here.